### PR TITLE
Add major as field and remove address as field

### LINKED ISSUE #16: Add major as field in student details
As a TA,
I can know the majors of my students,
So that I can understand their learning needs.

### LINKED ISSUE #30: Remove Address Property
Since ClassMonitor won't require to know a students address, that field can be deleted. Refactoring needs to be done to functional code and tests

### PR DESCRIPTION
**Pull request by kwuunnn**
Fix #16 and fix #30

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -25,13 +25,13 @@ public class AddCommand extends Command {
...
             + PREFIX_EMAIL + "EMAIL "
-            + PREFIX_ADDRESS + "ADDRESS "
+            + PREFIX_MAJOR + "MAJOR "
             + "[" + PREFIX_TAG + "TAG]...\n"
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_NAME + "John Doe "
             + PREFIX_PHONE + "98765432 "
             + PREFIX_EMAIL + "johnd@example.com "
-            + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
+            + PREFIX_MAJOR + "Computer Science"
```
Comment (chiaryan) [inline src/main/java/seedu/address/logic/commands/AddCommand.java]: Shld there be a trailing space here? `"Computer Science "`

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 2]
```diff
@@ -21,7 +21,7 @@ public class PersonCard extends UiPart<Region> {
      * As a consequence, UI elements' variable names cannot be set to such keywords
      * or an exception will be thrown by JavaFX during runtime.
      *
-     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue getMajor()Book level 4</a>
```
Comment (chiaryan) [inline src/main/java/seedu/address/ui/PersonCard.java]: Should be `on AddressBook`?

File: src/main/java/seedu/address/ui/PersonCard.java
Comment (chiaryan) [inline src/main/java/seedu/address/ui/PersonCard.java:1]: Idk how to comment on unchanged lines so I comment here

`private Label address;` shld be `private Label major;`? The corresponding field in `PersonListCard.fxml` shld be changed also.

File: src/main/java/seedu/address/model/person/Major.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ * Represents a Person's major in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidMajor(String)}
+ */
+public class Major {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Majors should be at least 2 characters long";
+
+    public static final String VALIDATION_REGEX = "(?<!\\s)[\\p{Alpha} ]{2,}";
```
Comment (chiaryan) [inline src/main/java/seedu/address/model/person/Major.java:33]: I'm not sure what the `(?<!\\s)` does here. I think it wld be good to document what this regex does.

**Review by chiaryan — COMMENTED** [review]
Overall good job on capturing most of the field changes. I've added some other comments also

File: src/main/java/seedu/address/logic/commands/AddCommand.java (hunk 1 above)
Comment (kwuunnn) [reply src/main/java/seedu/address/logic/commands/AddCommand.java]: Good catch! Thanks!

File: src/main/java/seedu/address/ui/PersonCard.java (hunk 2 above)
Comment (kwuunnn) [reply src/main/java/seedu/address/ui/PersonCard.java]: Good catch! Thanks!

File: src/main/java/seedu/address/ui/PersonCard.java
Comment (kwuunnn) [reply src/main/java/seedu/address/ui/PersonCard.java:1]: Good catch! Thanks!

File: src/main/java/seedu/address/model/person/Major.java (hunk 3 above)
Comment (kwuunnn) [reply src/main/java/seedu/address/model/person/Major.java:33]: Good catch! Thanks!

**Comment by evedaktyl** [comment]
LGTM

**Pull request merged**